### PR TITLE
Amazon Kindle block

### LINKED ISSUE #13427: Missing Amazon Kindle Instant Previews block
As of [#41615](https://core.trac.wordpress.org/changeset/41615), WordPress core supports embed of Amazon Kindle instant previews. However, there is no gutenberg block for this.

### PR DESCRIPTION
**Pull request by Soean**
## Description
Adds an Amazon Kindle embed block. 

Amazon Kindle embed support was added in WP 4.9 https://core.trac.wordpress.org/changeset/41615

Example URL: https://lesen.amazon.de/kp/embed?asin=B07HB5N1YN
How to get URLs: https://www.amazon.com/Kindle-Instant-Preview/b?ie=UTF8&node=13489836011

Fixes #13427 

kindle

kindle

### CODE REVIEW

File: packages/block-library/src/embed/core-embeds.js [hunk 1]
```diff
@@ -338,4 +339,25 @@ export const others = [
 		},
 		patterns: [ /^https?:\/\/wordpress\.tv\/.+/i ],
 	},
+	{
+		name: 'core-embed/amazon-kindle',
+		settings: {
+			title: 'Amazon Kindle',
+			icon: embedAmazonIcon,
+			keywords: [ __( 'ebook' ) ],
+			description: __( 'Embed Amazon Kindle content.' ),
```
Comment (notnownikki) [inline packages/block-library/src/embed/core-embeds.js]: we should add `responsive: false,` here too, as Amazon delivers the content in an iframe with fixed dimensions, but it doesn't conform to the usual video aspect ratios that the responsive styles use, so we get a load of extra padding that we don't want with responsive styles enabled.

**Review by notnownikki — CHANGES_REQUESTED** [review]
Just one request to change to the `responsive` setting to `false`, and this is looking good!

File: packages/block-library/src/embed/core-embeds.js (hunk 1 above)
Comment (Soean) [reply packages/block-library/src/embed/core-embeds.js]: Thanks for your feedback, I added it.

**Review by notnownikki — APPROVED** [review]

**Comment by notnownikki** [comment]
Thank you!

**Comment by karlcc58** [comment]
Hello, 
Thank you very much for your work.
It is possible to add a amazon affiliate code on the links? Which is the way?. 
Thanks and Greetings

**Comment by Soean** [comment]
@karlcc58 Yes, thats possible. See this step by step guide. On the last step you should have an option to select your affiliate code: https://www.amazon.com/Kindle-Instant-Preview/b?ie=UTF8&node=13489836011

**Comment by karlcc58** [comment]
Another question, It is possible to change "Buy on - Share - Free preview " for example with "Comprar en, compartir, Vista previa"
Again thank you very much :-)

**Comment by michaelebemis** [comment]
Can the background color be changed?

File: packages/block-library/src/embed/core-embeds.js [hunk 2]
```diff
@@ -338,4 +339,18 @@ export const others = [
...
+			title: 'Amazon Kindle',
+			icon: embedAmazonIcon,
+			keywords: [ __( 'ebook' ) ],
+			responsive: false,
+			description: __( 'Embed Amazon Kindle content.' ),
+		},
+		patterns: [
+			/^https?:\/\/([a-z0-9-]+\.)?(amazon|amzn)(\.[a-z]{2,4})+\/.+/i,
+			/^https?:\/\/(www\.)?(a\.co|z\.cn)\/.+/i,
+		],
```
Comment (mcsf) [inline packages/block-library/src/embed/core-embeds.js:354]: @Soean: This is too eagerly catching items that aren't books and attempting to render them as such.

Try pasting any product URL matching `https://www.amazon.com/gp/product/<product code>/` or `https://www.amazon.com/-/<locale>/dp/<product code>/`, for instance.

## Example

Pasting URL `https://www.amazon.com/dp/B0198CGQEE/` yields:

output

Comment (mcsf) [reply packages/block-library/src/embed/core-embeds.js:354]: Reported in #27547

**Pull request merged**